main.go

```diff
@@ -41,6 +49,11 @@ var (
 			Help: "Errors in requests to the SNMP exporter",
 		},
 	)
+	sc = &SafeConfig{
+		C: &config.Config{},
+	}
+	err      error
```

Comment:
Not the config, the `err`.  The error var needs to be used to the functions that use it, otherwise there is a global race condition.  Consider `var err error` or `err := foo` to replace this.